### PR TITLE
Bug 103137 - Maven Re-packaging

### PR DESCRIPTION
**Pull request by abhishekzambre**
Initial check-in for Maven Re-packaging

### CODE REVIEW

File: thirdparty/maven/Makefile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+# $(@D) == directory part of the target
+getsrc: $(psrc_file)
+$(psrc_file):
+	$(MKDIR) $(@D) && $(CD) $(@D) && $(WGET) $(purl)
+
+pkgadd: pkgadd_$(PKG_EXT)
+
+pkgadd_rpm:
+	$(PKG_EXTRACT) java-1.7.0-openjdk ant
```
Comment (plobbes) [inline thirdparty/maven/Makefile]: Wouldn't we prefer to use zimbra-openjdk here?

Comment (abhishekzambre) [inline thirdparty/maven/Makefile]: Oh yes, missed that. I was playing around and forgot to revert that change. Will make it now.

File: thirdparty/maven/zimbra-maven/debian/rules [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+%:
+	dh $@
+
+override_dh_auto_clean:
+
+override_dh_auto_configure:
+
+override_dh_strip:
+	dh_strip -pzimbra-maven
+	dh_strip -Nzimbra-maven
```
Comment (plobbes) [inline thirdparty/maven/zimbra-maven/debian/rules]: Hmm, what are these lines doing?  Seems like -p and -N are opposites... I'm wondering if this whole override is a waste, or if it's accomplishing something that's just non-obvious to me.

File: thirdparty/maven/zimbra-maven/debian/watch [hunk 3]
```diff
@@ -0,0 +1,2 @@
+version=3
+http://archive.apache.org/dist/maven/maven-3/3.3.9/source/apache-maven-(\d+\.\d+\.\d+).tar.xz
```
Comment (plobbes) [inline thirdparty/maven/zimbra-maven/debian/watch]: Hmm, ideally we don't have that hard coded 3.3.9 here.   Wonder if we want to pre-process the watch file to substitute something like PVER here.

File: thirdparty/maven/zimbra-maven/rpm/SPECS/maven.spec [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+%description
+The Zimbra maven build
+
+%prep
+%setup -n apache-maven-%{version}
+%patch0 -p1
+
+%build
+LDFLAGS="-Wl,-rpath,OZCL"; export LDFLAGS; \
+CFLAGS="-O2 -g"; export CFLAGS; \
```
Comment (plobbes) [inline thirdparty/maven/zimbra-maven/rpm/SPECS/maven.spec]: LDFLAGS and CFLAGS are probably bogus (unused, unnecessary) for maven, no?

File: thirdparty/maven/zimbra-maven/debian/watch (hunk 3 above)
Comment (plobbes) [inline thirdparty/maven/zimbra-maven/debian/watch]: Let's use https instead of http since it is available too.

File: thirdparty/maven/Makefile [hunk 5]
```diff
@@ -0,0 +1,71 @@
+PKG_ROOT := $(shell pwd)
+
+include $(PKG_ROOT)/../../package.def
+MAVEN_VERSION := 3.3.9
+pvers := $(MAVEN_VERSION)
+pname := maven
+pfile := apache-$(pname)-$(pvers)-src.tar.gz
+psrc_file := $(SRC_DIR)/$(pfile)
+purl := http://archive.apache.org/dist/maven/maven-3/$(pvers)/source/$(pfile)
```
Comment (plobbes) [inline thirdparty/maven/Makefile]: Use https instead of http here too please.

File: thirdparty/maven/zimbra-maven/debian/watch (hunk 3 above)
Comment (quanah) [inline thirdparty/maven/zimbra-maven/debian/watch]: Agreed, we use variables for substitution URLs that have the version in it in other Makefile's as well.

File: thirdparty/maven/zimbra-maven/debian/rules (hunk 2 above)
Comment (quanah) [inline thirdparty/maven/zimbra-maven/debian/rules]: Generally, I don't expect to see dh_strip used with java packages?

File: thirdparty/maven/zimbra-maven/debian/watch (hunk 3 above)
Comment (abhishekzambre) [inline thirdparty/maven/zimbra-maven/debian/watch]: Added "pvers" instead of hard coded version number, and url changed from "http" to "https".

**Comment by abhishekzambre** [comment]
Hi Quanah, Phil,

I have made changes as per suggestions.

File: thirdparty/maven/zimbra-maven/debian/control [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+Priority: optional
+Maintainer: Zimbra Packaging Services <packaging-devel@zimbra.com>
+Uploaders: Zimbra Packaging Services <packaging-devel@zimbra.com>
+Standards-Version: 3.9.5
+Homepage:  https://maven.apache.org/
+
+Package: zimbra-maven
+Priority: optional
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends} zimbra-openjdk (= ${binary:Version})
```
Comment (quanah) [inline thirdparty/maven/zimbra-maven/debian/control]: Should just depend on zimbra-openjdk, not a specific version of it.

File: thirdparty/maven/zimbra-maven/debian/control [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+Priority: optional
+Maintainer: Zimbra Packaging Services <packaging-devel@zimbra.com>
+Uploaders: Zimbra Packaging Services <packaging-devel@zimbra.com>
+Standards-Version: 3.9.5
+Homepage:  https://maven.apache.org/
+
+Package: zimbra-maven
+Priority: optional
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends} zimbra-openjdk
```
Comment (quanah) [inline thirdparty/maven/zimbra-maven/debian/control]: I believe you're missing a comma after Depends}

I.e., it should be Depends}, zimbra-openjdk

Comment (abhishekzambre) [inline thirdparty/maven/zimbra-maven/debian/control]: Corrected.

**Comment by quanah** [comment]
+1 looks good to me.

File: thirdparty/maven/zimbra-maven/rpm/SPECS/maven.spec [hunk 8]
```diff
@@ -0,0 +1,30 @@
+Summary:            Zimbra's maven build
+Name:               zimbra-maven
+Version:            VERSION
+Release:            ITERATIONZAPPEND
+License:            Apache-2.0
+Source:             %{name}-%{version}.tar.gz
+Patch0:             skip_rat_build_xml.patch
+BuildRequires:		zimbra-openjdk
```
Comment (plobbes) [inline thirdparty/maven/zimbra-maven/rpm/SPECS/maven.spec]: nit: tabs => spaces on lines 8 and 9

**Comment by plobbes** [comment]
+1 other than the minor whitespace issue noted above, LGTM

**Comment by abhishekzambre** [comment]
Merge completed.

**Pull request merged**